### PR TITLE
Issue 4151: Add option to make RocksDB only use memory environment

### LINKED ISSUE #4151: Add an option for RocksDB to use memory only environment
**Problem description**
Let RocksDB use memory-only environment will help us analyze whether RocksDB cache disk is the performance bottleneck

**Problem location**
RocksDB config

**Suggestions for an improvement**
Add an option

### PR DESCRIPTION
**Pull request by Tristan1900**
Signed-off-by: wenqimou <452787782@qq.com>

**Change log description**  
This PR adds an option for Pravega to enable RocksDB use memory-only environment. 

**Purpose of the change**  
Fix #4151 

**What the code does**  
There is a configuration in RocksDB java client called `RocksMemEnv` that the code uses to enable the memory-only environment.

**How to verify it**  
Deploy a cluster using such image, run a workload and monitor the disk of the segmentstore cache. There will be nothing change as all the `.sst` files are not flushed to the disk.

### CODE REVIEW

File: segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/rocksdb/RocksDBCache.java [hunk 1]
```diff
@@ -257,6 +261,12 @@ private Options createDatabaseOptions() {
                 .setTableFormatConfig(tableFormatConfig)
                 .setOptimizeFiltersForHits(true)
                 .setUseDirectReads(this.directReads);
+
+        if(this.memoryOnly) {
```
Comment (RaulGracia) [inline segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/rocksdb/RocksDBCache.java]: Minor: missing space after `if`.

File: segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/rocksdb/RocksDBConfig.java [hunk 2]
```diff
@@ -69,6 +70,13 @@
     @Getter
     private final boolean directReads;
 
+    /**
+     * Enabling memory only will let rocksdb write everything into memory instead of disks. This supposed to provide
```
Comment (RaulGracia) [inline segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/rocksdb/RocksDBConfig.java]: Minor: `rocksdb` -> `RocksDB`, `This (is) supposed`.

File: segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/rocksdb/RocksDBConfig.java [hunk 3]
```diff
@@ -69,6 +70,13 @@
...
+     * very high performance since there is not disk IO incurred.
+     */
+    @Getter
+    private final boolean memoryOnly;
```
Comment (RaulGracia) [inline segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/rocksdb/RocksDBConfig.java]: We need to add this property and the default value in the `config.properties` file.

File: segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/rocksdb/RocksDBCache.java (hunk 1 above)
Comment (Tristan1900) [reply segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/rocksdb/RocksDBCache.java]: Got it. Thanks!

File: segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/rocksdb/RocksDBConfig.java (hunk 2 above)
Comment (Tristan1900) [reply segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/rocksdb/RocksDBConfig.java]: Got it

File: segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/rocksdb/RocksDBConfig.java (hunk 3 above)
Comment (Tristan1900) [reply segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/rocksdb/RocksDBConfig.java]: Added to the properties. Thanks!

**Comment by RaulGracia** [comment]
@tkaitchuck please could you have a look to this PR?

**Comment by Tristan1900** [comment]
@andreipaduroiu Yes, I have added a note saying this is for experiment use only. Thanks

**Comment by RaulGracia** [comment]
@andreipaduroiu the RocksDB in-memory config option has been tagged as experimental and "do not use in production". Is that ok for you for merging this?

**Comment by andreipaduroiu** [comment]
I will approve this since it provides sensible improvements to performance. But it may have a short life as it will likely be replaced with #4044.

**Comment by RaulGracia** [comment]
@fpj do you think that this is ready to go in?

**Comment by RaulGracia** [comment]
@fpj can you please merge this PR when possible?

**Pull request merged**